bin/npg_publish_tree.pl

```diff
@@ -188,46 +188,32 @@ sub handler {
 my @files = grep { -f } $publisher->list_directory($source_directory,
                                                    recurse => 1);
 
-my $mlwh_json_cb = sub {
-  # DataObject, path to file folder, collection file
-  my ($obj, $collection, $file) = @_;
-  if (defined $mlwh_json_filename) { 
+sub write_json {
+  my ($json_filename, $irods_collection) = @_;
+  if (defined $json_filename && (not $json_filename =~ qr/^\s*$/sxm)
+      && defined $irods_collection && (not $irods_collection =~ qr/^\s*$/sxm)) {
     Readonly::Scalar my $JSON_FILE_VERSION => '1.0';
```

Comment:
Uppercase is often used for package-scope "static" variables. These should be at the top of the file (see `VERSION`, which is already there)

Reply:
Yes, just seen it. I will move the variable close to the other.